### DISCUSSION TITLE
Official Node APIs

### DISCUSSION THREAD

**Discussion by Rasika9596** [dogecoin/dogecoin#2388] [Q&A]
Hello,

I got this JSON APIs - [https://dogechain.info/api/blockchain_api](url). 
Does these official APIs? Will it provide long time support? Are you giving any guarantee that it will be up and running every time? Internally which IP/Node is hitting? 

Actually, I want to use it for the long term if these are official ones on which I can trust. So that I want confirmation on these points. Please help me by letting this know.

With this can anyone please provide me the official node APIs documentation which will work on our own node? I want APIs to fetch block information, transactions details in it, etc. I'm not able to find such a document so, please help.

**Comment by AbcSxyZ** [comment]
Hi,

Definitively, it's not an official API. Nodes are the network, they have the information and you should query them instead.
`dogecoind` implement the dogecoin(bitcoin) protocol within the core, and you have RPC (Json-RPC) interface to query information.

Currently, the dogecoin documentation need a rework (see [Docs4Doge](https://github.com/dogecoin/dogecoin/issues/2271)), there is no information related to RPC. It is a thing to do, but until it's available, you should definitively take a look to some bitcoin documentation. Some commands may be different/unavailable, but should almost be the same.

Idk what is the best resource, I can suggest you to check the following to start : https://developer.bitcoin.org/reference/rpc/

You can host your node and query it, or perhaps (you can try I'm not sure) you can find an existing node and query it directly using [DNS seeds](https://www.oreilly.com/library/view/blockchain-developers-guide/9781789954722/ccb46585-403e-44dd-a0a9-253bb48f2736.xhtml) available [here](https://github.com/dogecoin/dogecoin/blob/e34197907f32f3fca371bfc490ef6e521723a070/src/chainparams.cpp#L164).

**Comment by Rasika9596** [comment]
Thanks for the details! We'll go through the bitcoin documentation and will test with the required APIs. 
But still, we'll be waiting for this :)

**Comment by patricklodder** [comment]
Most of the explorer APIs have been around for a long time. 

For interacting with your own Dogecoin Core node, you can use [jayson](https://www.npmjs.com/package/jayson)

You can use [bitcoinjs-lib](https://www.npmjs.com/package/bitcoinjs-lib) with [these params](https://github.com/BlockIo/block_io-nodejs/blob/081caf2af1b6d318d490fc73feb66df90e7c7512/data/networks.js#L15-L38) for transaction creation.

**Comment by fmhc** [comment]
@AbcSxyZ can we have a section in the Docs4Doge project with a listing for doge related software projects? There are quite some things out there, would love to help maintaining a list of community projects / useful APIs / Libraries around Dogechain

**Comment by AbcSxyZ** [comment]
@fmhc I clearly agree with the idea of listing community projects, one item I added in Docs4Doge is `Link to community projects`.

The "community projects" is a vague topic regarding Dogecoin, it could be technical projects related to the core like libraries, or various type of projects like application of the ecosystem (wallets, extensions, bot...), charity project, merch... Idk how all those things should be handled and integrated (or not), the treatment may differ depending on the project, I would love to see the topic discussed and improvement in this area.

Would be great for various reason, to let blockchain developer creating applications and interact with the core, to share and valorize new projects, to give visibility, to reinforce the Dogecoin ecosystem :)
But come with some downside (scam...), so idk what is the current "repository policy".

Maybe starting a new discussion would be a great point to start, to see what need/could/should be integrated, which form, etc.  
If you want to start with what you've in mind, I will definitively follow and support the idea :heart:
